examples/acoustics/acoustic.py

```diff
@@ -0,0 +1,179 @@
+    pressure_data = pd.read_csv(acsv,skipinitialspace=True)
+    times = pressure_data['physical_time']
+    steps = pressure_data['physical_step']
+    p = pressure_data.iloc[:,2:-1]
+    return [np.array(times),np.array(steps),np.array(p)]
+
+def filterZeroAcousticData(time,steps,pressure):
+    numObservers = pressure.shape[-1]
+    acoustic_data_indices = {}
+    acoustic_data = {}
+    # loop over observers to find where they are zero
+    for i in range(numObservers):
+        p_observer = pressure[:,i]
+        acoustic_data_indices[i] = np.where(p_observer!=0)[0]
+    first_elements = []
+    last_elements = []
+    # lists of elements with zeros at begin and end
+    for i in range(numObservers):
+        first_elements.append(acoustic_data_indices[i][0])
+        last_elements.append(acoustic_data_indices[i][-1])
+    # max index where still there is a zero element for all observers
+    start_index = max(first_elements)
+    # min index where still there is a zero element for all observers
+    end_index = min(last_elements)
+
+    # physical time and step where all observers all non zero
+    acoustic_data['physical_time'] = time[start_index:end_index]
+    acoustic_data['physical_step'] = steps[start_index:end_index]
+
+    # non-zero acoustic data for all observers with the same physical time
+    for i in range(numObservers):
+        p_observer = pressure[:,i]
+        acoustic_data[i] = p_observer[start_index:end_index]
+    return acoustic_data
+
+def cal_spl(signal_amp):
+    p_ref=2e-5
+    return 10*np.log10(signal_amp/p_ref)
+
+def plot_response(name,freq,resp,key):
+    fig, ax = plt.subplots()
+    # optional axis limit
+    ax.axis([100, 18000, 0, 70])
+    plt.plot(freq,resp,'crimson',linewidth=0.8,marker='o',markersize=0.8)
+    plt.xscale('log')
+    for axis in [ax.xaxis, ax.yaxis]:
+        formatter = ticker.ScalarFormatter()
+        formatter.set_scientific(False)
+        axis.set_major_formatter(formatter)
+    ax.yaxis.set_minor_formatter(formatter)
+    ax.xaxis.set_minor_formatter(formatter)
+    
+    # optional log minor ticker
+    ax.xaxis.set_minor_locator(ticker.LogLocator(base=10.0, subs=(0.2, 0.4, 0.6, 0.8), numticks=5))
+
+    plotTitle = f'Noise Spectrum - Observer {key}' if key != 'avg' else f'Noise Spectrum - Averaged'
+    plt.title(plotTitle, fontsize=10)
+    plt.xlabel('Frequency [Hz]', fontsize=10)
+    plt.ylabel('SPL [dB]', fontsize=10)
+    plt.grid(which='major')
+    plt.grid(which='minor', ls='--',color='k',linewidth=0.5)
+    ax.tick_params(axis='x',which='major',labelsize=8,pad=10)
+    ax.tick_params(axis='y',which='major',labelsize=8)
+    ax.tick_params(axis='both',which='minor',labelsize=5)
+    
+    # figure resolution
+    plt.rcParams['figure.dpi'] = 500
+    plt.rcParams['savefig.dpi'] = 500
+    plt.savefig(f'{name}_narrow-band_spectra_{key}.png', bbox_inches='tight')
+    plt.close()
+
+def getSpectrum(aSound,rho,acsv,oName):
```

Comment:
stick to one case type: snake_case or camelCase, python users generally prefer snake_case